src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.scss']
+})
+export class RegistrationComponent implements OnInit {
+
+  public firstName: string;
+  public lastName: string;
+  public email: string;
+  public password: string;
+  public passwordConfirmation: string;
+
+  private passwordRegex = new RegExp(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/);
+  // tslint:disable-next-line: max-line-length
+  private emailRegex = new RegExp(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  public sendRegistrationRequest(): void {
+    if (!this.errorsInForm()) {
+
+    }
+  }
+
+  private errorsInForm(): boolean {
+    // TODO: visually show errors
+    let errors = false;
+    if (this.firstName == null) {
+      alert('Il nome è necessario');
+      errors = true;
+    }
+    if (this.lastName == null) {
+      alert('Il cognome è necessario');
+      errors = true;
+    }
+    console.log(this.email);
+    if (this.email == null) {
+      alert('L\'email è necessaria');
+      errors = true;
+    } else if (!this.emailRegex.test(this.email)) {
+      alert('L\'email non è valida');
```

Comment:
scriviamo una issue per questi `alert()`, del tipo "Remove alert() from the project" ... :S